src/main/java/com/nerodesk/App.java

```diff
@@ -100,113 +102,121 @@ public App(final Base base) throws IOException {
+                        @Override
+                        public Take route(final Request req)
+                            throws IOException {
+                            final String file =
+                                new RqHref(req).href().param("x")
+                                    .iterator().next();
+                            return new TkDelete(
+                                base.user(
+                                    new RqAuth(req).identity().urn()
+                                ).docs().doc(file)
+                            );
                         }
-                    )
-                )
-            ),
-            new FkRegex(
-                "/r",
-                new Takes() {
-                    @Override
-                    public Take route(final Request req) throws IOException {
-                        final String file =
-                            new RqHref(req).href().param("f").iterator().next();
-                        return new TkRead(
-                            base.user(
-                                new RqAuth(req).identity().urn()
-                            ).docs().doc(file)
-                        );
                     }
-                }
-            ),
-            new FkRegex(
-                "/d",
-                new Takes() {
-                    @Override
-                    public Take route(final Request req) throws IOException {
-                        final String file =
-                            new RqHref(req).href().param("x").iterator().next();
-                        return new TkDelete(
-                            base.user(
-                                new RqAuth(req).identity().urn()
-                            ).docs().doc(file)
-                        );
+                ),
+                new FkRegex(
+                    "/w",
+                    new Takes() {
+                        @Override
+                        public Take route(final Request req)
+                            throws IOException {
+                            return new TkWrite(
+                                base.user(
+                                    new RqAuth(req).identity().urn()
+                                ).docs(),
+                                req
+                            );
+                        }
                     }
-                }
+                )
             ),
-            new FkRegex(
-                "/w",
-                new Takes() {
-                    @Override
-                    public Take route(final Request req) throws IOException {
-                        return new TkWrite(
-                            base.user(
-                                new RqAuth(req).identity().urn()
-                            ).docs(),
-                            req
-                        );
-                    }
-                }
-            )
+            new TkHTML("oops, something went wrong!")
```

Comment:
@longtimeago well... this is not really elegant. every time we'll see this problem - we won't know what happened. we need to show a stacktrace here somehow. or save it somewhere. or email it to someone. or at least log it. but now you're just swallowing all exceptions of the application. it's a bad idea.